steps/25_pretrain/step.md

```diff
@@ -0,0 +1,244 @@
+* pre-training
+* fine-tuning of foundation models
+* zero-shot use of foundation models
+
+References:
+
+* conceptual design issue: https://github.com/sktime/sktime/issues/6580
+* umbrella issue foundation models: https://github.com/sktime/sktime/issues/6177
+* newer issue: https://github.com/sktime/sktime/issues/7838
+
+### requirements
+
+* design covers the above use cases with a simple interface
+* composability - use of sensible pipelines, tuning, etc should be simple and not require major surgery in current compositors
+* downwards and upwards compatibility - design should not impact current extension contracts
+* maintainability: maintaining the framework and estimators with the above capabilities should be simple
+
+### The proposed solution
+
+Our proposed solution adds a new state, and a simple switch.
+
+No new public methods are added beyond this, and signatures of methods are not modified.
+
+Estimators get a third state, "pretraining phase", besides unfitted and fitted.
+
+The solution is best illustrated in the basic vignette below.
+
+### Discussion of current solutions
+
+There are multiple current solutions, all have problems:
+
+#### Global forecasting
+
+Forecasters inheriting from `_BaseGlobalForecaster`.
+
+A `y` is added in the `predict` methods. If this is passed, the `fit` is interpreted
+as a pretraining pass.
+
+Problems:
+
+* some models need to know at `fit` time whether the data is for pretraining.
+Examples: global reduction approaches. Broadcasting.
+* as a general design principle, all `predict` methods would need to be changed to
+allow for addition of the `fit` arguments. This clashes in cases where arguments
+of the same name are present both in `fit` and `predict`, e.g., the `X` in forecasting,
+or all arguments for transformations.
+
+#### Pre-training foundation models
+
+Foundation models currently come in two different, contradictory forms:
+
+* those that carry out fine-tuning in `fit`
+* those that pass the context in `fit`, e.g., zero-shot models
+
+Problems: This is inconsistent, and it does not seem to be possible - without an `__init__` arg
+that acts as a switch, or in different classes, to have the same weights in the same class
+be part of a zero-shot or fine-tuning algorithm.
+
+
+## Design: pretraining vignette
+
+Presenting user facing API. For delineation against current designs:
+
+* no new arguments are added to `predict`-like methods
+* a flag is added before or at `fit` to determine whether usage is normal fitting, or pre-training.
+    * two vignettes are presented that pass this information on, for discussion.
+
+### basic usage vignette
+
+Illustrated for global forecasting.
+
+```python
+
+y_pretrain, X_pretrain = load_pretrain_data()
+
+f = MyForecasterCls(params)
+
+f.pretrain()
+
+f.fit(y=y_pretrain, X=X_pretrain)
+
+# fh is optional, but some models require this
+
+f.pretrain("off")
+
+# usual vignette starts here
+y, X = load_data()
+
+f.fit(y, X, fh=[1, 2, 3])
```

Comment:
What would do this call? Is it finetuning or only providing context? This is not clear to me.

Reply:
this is abstract. It could in theory involve ephemeral fine-tuning, but normally would only pass the context, for foundation models.

Reply:
I added it in some other thread. I think we need to collect the different types of `fit`.